app/screens/send-bitcoin-screen/payment-details/index.types.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  GraphQlApplicationError,
+  PaymentSendResult,
+  useIntraLedgerPaymentSendMutation,
+  useIntraLedgerUsdPaymentSendMutation,
+  useLnInvoiceFeeProbeMutation,
+  useLnInvoicePaymentSendMutation,
+  useLnNoAmountInvoiceFeeProbeMutation,
+  useLnNoAmountInvoicePaymentSendMutation,
+  useLnNoAmountUsdInvoiceFeeProbeMutation,
+  useLnNoAmountUsdInvoicePaymentSendMutation,
+  useLnUsdInvoiceFeeProbeMutation,
+  useOnChainPaymentSendMutation,
+  useOnChainTxFeeLazyQuery,
+  WalletCurrency,
+} from "@app/graphql/generated"
+import { BtcPaymentAmount, PaymentAmount } from "@app/types/amounts"
+import { WalletDescriptor } from "@app/types/wallets"
+import { LnUrlPayServiceResponse } from "lnurl-pay/dist/types/types"
+
+export type ConvertPaymentAmount = <T extends WalletCurrency>(
+  paymentAmount: PaymentAmount<WalletCurrency>,
+  toCurrency: T,
+) => PaymentAmount<T>
+
+export type BaseCreatePaymentDetailsParams<T extends WalletCurrency> = {
+  convertPaymentAmount: ConvertPaymentAmount
+  sendingWalletDescriptor?: WalletDescriptor<T>
```

Comment:
in which scenario is `sendingWalletDescriptor` optional?

Reply:
UseFee and SendPayment will not be defined untill sendingWalletDescriptor is defined, however you could imagine a scenario where you would create the PaymentDetail, then set the amount, then set the sendingWalletDescriptor.